test/xkeyboard-config-test.py.in

```diff
@@ -209,6 +221,22 @@ def run(combos, tool, njobs):
             if target:
                 print(invocation, file=target)
 
+            if keymap_output_dir:
+                # we're running through the layouts in a somewhat sorted manner,
+                # so let's keep the fd open until we switch layouts
+                layout = invocation.layout
+                if invocation.variant:
+                    layout += f"({invocation.variant})"
+                fname = keymap_output_dir / layout
+                if fname != keymap_file:
+                    keymap_file = fname
+                    keymap_file_fd = open(keymap_file, 'a')
```

Comment:
Probably should  close the old one first

Reply:
fixed, thanks